scripts/test_backref_cost_attribution_v14_protocol.py

```
@@ -0,0 +1,306 @@
+    brace = text.index("{", start)
+    depth = 0
+    for index in range(brace, len(text)):
+        depth += (text[index] == "{") - (text[index] == "}")
+        if depth == 0:
+            return text[brace:index + 1]
+    raise AssertionError(f"unterminated function: {name}")
+
+
+def semantic_identity() -> None:
+    frozen = subprocess.check_output(
+        ["git", "show", "673bc3fa:src/enc/backward_references_cost_enc.c"],
```

Ruff (0.16.1)

[error] Starting a process with a partial executable path

(S607)